### PR TITLE
Expand the default rancher-resource-set 

### PR DESCRIPTION
**Pull request by mrajashree**
Expand the rancher-resource-set to include
1. All new 2.5 CRDs such as catalog.cattle.io
2. [eks-operator](https://github.com/rancher/system-charts/tree/dev-v2.4/charts/rancher-eks-operator) 
I'm able to perform restore with eks-cluster  running. If I deploy a hosted EKS cluster, and then migrate rancher to a new cluster, the user EKS cluster gets restored properly. I do see the eks-operator pod running post restore. I could also deploy another eks cluster after that

https://github.com/rancher/backup-restore-operator/issues/6

### CODE REVIEW

File: packages/backup-restore-operator/charts/templates/rancher-resourceset.yaml [hunk 1]
```diff
@@ -17,16 +17,32 @@ resourceSelectors:
...
+      - "eks-operator"
   - apiVersion: "rbac.authorization.k8s.io/v1"
     kindsRegexp: "^clusterroles$"
     resourceNameRegexp: "^cattle-|^p-|^c-|^local-|^user-|^u-|^project-|^create-ns$"
+    resourceNames:
+      - "eks-operator"
   - apiVersion: "apiextensions.k8s.io/v1beta1"
     kindsRegexp: "."
-    resourceNameRegexp: "management.cattle.io$|project.cattle.io$"
+    resourceNameRegexp: "management.cattle.io$|project.cattle.io$|catalog.cattle.io$|eks.cattle.io$|resources.cattle.io$"
```
Comment (aiyengar2) [inline packages/backup-restore-operator/charts/templates/rancher-resourceset.yaml:29]: How are we handling 2.5 CRDs such as Monitoring, Istio, Logging etc. that are in different resource groups like `monitoring.coreos.com`? Will feature charts be expected to create a resourceset on chart deploy similar to how monitoring expects feature charts to create ServiceMonitors?

**Review by aiyengar2 — APPROVED** [review]
LGTM

File: packages/backup-restore-operator/charts/templates/rancher-resourceset.yaml (hunk 1 above)
Comment (mrajashree) [reply packages/backup-restore-operator/charts/templates/rancher-resourceset.yaml:29]: yes we will have to create ResourceSets per 2.5 feature chart.

**Review by rmweir — CHANGES_REQUESTED** [review]
Is the service account "eks-operator" being restored? https://github.com/rancher/system-charts/blob/dev-v2.4/charts/rancher-eks-operator/0.1.0/templates/serviceaccount.yaml

**Comment by mrajashree** [comment]
@rmweir yes, since the eks-operator service account is in cattle-system namespace, it gets backed up with [this selector](https://github.com/rancher/charts/blob/0dfe29e42fd404463cb7fb2b62a441cb9742cae8/packages/backup-restore-operator/charts/templates/rancher-resourceset.yaml#L13)

**Comment by rmweir** [comment]
@mrajashree I didn't realize that is a namespace regex. Super cool!

**Review by rmweir — APPROVED** [review]
LGTM

File: packages/backup-restore-operator/charts/templates/rancher-resourceset.yaml [hunk 2]
```diff
@@ -17,16 +17,32 @@ resourceSelectors:
   - apiVersion: "rbac.authorization.k8s.io/v1"
     kindsRegexp: "^clusterrolebindings$"
     resourceNameRegexp: "^cattle-|^clusterrolebinding-|^globaladmin-user-|^grb-u-"
+    resourceNames:
+      - "eks-operator"
   - apiVersion: "rbac.authorization.k8s.io/v1"
     kindsRegexp: "^clusterroles$"
     resourceNameRegexp: "^cattle-|^p-|^c-|^local-|^user-|^u-|^project-|^create-ns$"
+    resourceNames:
+      - "eks-operator"
```
Comment (cbron) [inline packages/backup-restore-operator/charts/templates/rancher-resourceset.yaml:26]: So if it matches either `resourceNameRegexp` or `resourceNames` it catches it ?

**Review by cbron — COMMENTED** [review]
Need to updated conflicts then looks good

File: packages/backup-restore-operator/charts/templates/rancher-resourceset.yaml (hunk 2 above)
Comment (mrajashree) [reply packages/backup-restore-operator/charts/templates/rancher-resourceset.yaml:26]: yes, it's an OR, `resourceNameRegexp || resourceNames`

**Review by cbron — APPROVED** [review]

**Pull request merged**